File: trunk/ruby/fetchmail.rb
#!/usr/bin/env ruby
require 'net/imap'
#require 'mail'

username = 'derbymonsanto.bot@gmail.com'
password = 'gdm12345-'

imap = Net::IMAP.new('imap.gmail.com','993',true)
imap.login(username, password)
imap.select('INBOX')
  imap.search(["NOT", "DELETED"]).each do |message_id|
    envelope = imap.fetch(message_id, "ENVELOPE")[0].attr["ENVELOPE"]
    puts "#{envelope.from[0].name}: \t#{envelope.subject}"
  end
imap.logout()
imap.disconnect()

File: trunk/python/gmail/fetchmail.py
#!/usr/bin/python
import libgmail

#ga = libgmail.GmailAccount("derbymonsanto.bot@gmail.com", "gdm12345-")
ga = libgmail.GmailAccount("guillaume.riflet@gmail.com", "qwerty12345-")
ga.login()
folder = ga.getMessagesByFolder('inbox')

for thread in folder:
    print thread.id, len(thread), thread.subject
    for msg in thread:
        print "  ", msg.id, msg.number, msg.subject
        print msg.sourcerequire 
